### PR TITLE
Cap the Loki container requests to 50m of CPU and 300Mi of Memory for the test-machinery

### LINKED ISSUE #4747: [Flaky test] `should get container logs from loki by tenant` test is flaky
**How to categorize this issue?**

/area quality
/area testing
/kind bug
/priority 3

**What happened**:
The success rate of the beta tests is awful:

Screenshot 2021-09-29 at 19 39 49

One of the most prominent failures comes from the `should get container logs from loki by tenant` with reason:

```
Seed logging testing
/src/test/integration/shoots/logging/loki_tenant.go:61
  
  /src/test/framework/test_options.go:56
    [BETA] [SHOOT] should get container logs from loki by tenant [It]
    /src/test/framework/gingko_utils.go:26

    Unexpected error:
        <*retry.Error | 0xc0005b9300>: {
            ctxError: <context.deadlineExceededError>{},
            err: <*errors.errorString | 0xc000d7ea20>{
                s: "received only 0/100 logs",
            },
        }
        retry failed with context deadline exceeded, last error: received only 0/100 logs
    occurred

    /src/test/integration/shoots/logging/loki_tenant.go:175
```

**What you expected to happen**:
No test failures

**How to reproduce it (as minimally and precisely as possible)**:
Has to be investigated.

**Environment**:

- Gardener version: v1.32.1
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

### PR DESCRIPTION
**Pull request by vlvasilev**
**How to categorize this PR?**

/area logging
/kind test

**What this PR does / why we need it**:
This PR caps the Loki container requests to 50m of CPU and 300Mi of Memory when executing the `should get container logs from loki for all namespaces` test of the test-machinery.  This is needed because when we run the tests on two nodes of 4Gi of Memory and the seed central Loki container is scaled up to 3Gi of memory there is no enough free memory for Loki pods to be scheduled.  

**Which issue(s) this PR fixes**:
Partially fixes [#4747](https://github.com/gardener/gardener/issues/4747)

**Special notes for your reviewer**:

**Release note**:

```bugfix developer
The memory and CPU requests of the loki containers are capped to 50mi of CPU and 300Mi of memory for the logging test-machinery tests.
```

### CODE REVIEW

**Review by Kristian-ZH — APPROVED** [review]
/lgtm

File: test/testmachinery/shoots/logging/seed_logging_stack.go [hunk 1]
```diff
@@ -397,9 +401,28 @@ func prepareShootLokiStatefulSet(shootLokiSts, gardenLokiSts *appsv1.StatefulSet
...
 	// Move the shoot Loki in the garden namespace
 	shootLokiSts.Namespace = v1beta1constants.GardenNamespace
+
+	capLokiMemoryAndCPURequests(shootLokiSts)
+
 	return shootLokiSts
 }
 
+func capLokiMemoryAndCPURequests(shootLokiSts *appsv1.StatefulSet) {
+	for ind, container := range shootLokiSts.Spec.Template.Spec.Containers {
```
Comment (ialidzhikov) [inline test/testmachinery/shoots/logging/seed_logging_stack.go]:
```suggestion
	for i, container := range shootLokiSts.Spec.Template.Spec.Containers {
```

File: test/testmachinery/shoots/logging/seed_logging_stack.go [hunk 2]
```diff
@@ -397,9 +401,28 @@ func prepareShootLokiStatefulSet(shootLokiSts, gardenLokiSts *appsv1.StatefulSet
...
+		if container.Name == "loki" {
+			shootLokiSts.Spec.Template.Spec.Containers[ind].Resources = corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("50m"),
+					corev1.ResourceMemory: resource.MustParse("300Mi"),
+				},
+				Limits: corev1.ResourceList{
+					corev1.ResourceMemory: resource.MustParse("3Gi"),
+				},
+			}
```
Comment (ialidzhikov) [inline test/testmachinery/shoots/logging/seed_logging_stack.go:421]: You can break at this point, right?

File: test/testmachinery/shoots/logging/seed_logging_stack.go [hunk 3]
```diff
@@ -397,9 +401,28 @@ func prepareShootLokiStatefulSet(shootLokiSts, gardenLokiSts *appsv1.StatefulSet
 	shootLokiSts.Spec.Template.Labels = newLabels
 	// Move the shoot Loki in the garden namespace
 	shootLokiSts.Namespace = v1beta1constants.GardenNamespace
+
+	capLokiMemoryAndCPURequests(shootLokiSts)
+
 	return shootLokiSts
 }
 
+func capLokiMemoryAndCPURequests(shootLokiSts *appsv1.StatefulSet) {
```
Comment (ialidzhikov) [inline test/testmachinery/shoots/logging/seed_logging_stack.go]: It is not only the Shoot loki, you can the same func for the garden/central loki.

```suggestion
func capLokiMemoryAndCPURequests(lokiSts *appsv1.StatefulSet) {
```

Comment (ialidzhikov) [inline test/testmachinery/shoots/logging/seed_logging_stack.go]: The func is called `capLokiMemoryAndCPURequests` but it is not only reseting the requests, but also setting a memory limit. Probably a better would be `capLokiContainerResources`/`resetLokiContainerResources`. WDYT?

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/approve
/milestone v1.54

**Pull request merged**